### ISSUE TITLE
Add Instance norm?

### ISSUE THREAD

**Issue by DmitryUlyanov** [pytorch/pytorch#1215]
Hello, what about adding InstanceNorm2d layer to the core? It is used in stylization and found its place in GANs. 

Will it be OK if it is implemented using BatchNorm2d as in [torch code](https://github.com/DmitryUlyanov/texture_nets/blob/master/InstanceNormalization.lua)? 

BarchNorm2d can be avoided though. Found this implementation by @darkstar112358 [here](https://github.com/darkstar112358/fast-neural-style/blob/master/neural_style/transformer_net.py#L111), which does not use BatchNorm. Which one do you prefer? I would be able to make a PR then.

**Comment by soumith** [comment]
it would likely have better performance using BatchNorm, because we have fused kernels for that.

**Comment by darkstar112358** [comment]
@DmitryUlyanov a side-note: in the experiments I ran, my `InstanceNormalization` code was almost as fast as `BatchNorm2d`.

**Comment by fmassa** [comment]
This is already in master
